demos/python_demos/image_inpainting_demo/image_inpainting_demo.py

```diff
@@ -33,30 +34,103 @@ def build_argparser():
+            brush_width = np.random.randint(10, max_brush_width + 1) // 2 * 2
+            next_y = start_y + length * np.cos(angle)
+            next_x = start_x + length * np.sin(angle)
+
+            next_y = np.clip(next_y, 0, h - 1).astype(np.int)
+            next_x = np.clip(next_x, 0, w - 1).astype(np.int)
+            cv2.line(mask, (start_y, start_x), (next_y, next_x), 1, brush_width)
+            cv2.circle(mask, (start_y, start_x), brush_width // 2, 1)
+
+            start_y, start_x = next_y, next_x
+    return mask
+
+
+def inpaintRandomHoles(args):
+    if len(args.mask_color) != 3:
+        print("Invalid mask_color is provided. Please provide 3 RGB components\n")
+        exit()
+    mask_color=args.mask_color[::-1] # argument comes in RGB mode, but we will use BGR notation below
 
     ie = IECore()
-    inpainting_processor = ImageInpainting(ie, args.model, args.parts,
-                                           args.max_brush_width, args.max_length, args.max_vertex, args.device)
+    img = loadImage(args.input)
+
+    inpainting_processor = ImageInpainting(ie, args.model, args.device)
+
+    #--- Resize to model input and generate random mask
+    img = cv2.resize(img, (inpainting_processor.input_width,inpainting_processor.input_height))
 
-    img = cv2.imread(args.input, cv2.IMREAD_COLOR)
-    masked_image, output_image = inpainting_processor.process(img)
+    if args.rnd:
+        mask = createRandomMask(args.parts,args.max_vertex,args.max_length,args.max_brush_width,
+                                inpainting_processor.input_height,inpainting_processor.input_width)
+    else:
+        top = np.full(img.shape,mask_color,np.uint8)
+        mask = cv2.inRange(img,top,top) / 255
+        mask = np.expand_dims(mask,2)
+
+    masked_image = (img * (1 - mask) + 255 * mask).astype(np.uint8)
+
+    #--- Inpaint and show results
+    output_image = inpainting_processor.process(masked_image,mask)
     concat_imgs = np.hstack((masked_image, output_image))
-    cv2.putText(concat_imgs, 'summary: {:.1f} FPS'.format(
-            float(1 / inpainting_processor.infer_time)), (5, 15), cv2.FONT_HERSHEY_COMPLEX, 0.5, (0, 0, 200))
-    if not args.no_show:
-        cv2.imshow('Image Inpainting Demo', concat_imgs)
-        cv2.waitKey(0)
+    cv2.putText(concat_imgs, 'summary: {:.1f} FPS'.format(float(1 / inpainting_processor.infer_time)), (5, 15), cv2.FONT_HERSHEY_COMPLEX, 0.5, (0, 0, 200))
+    return concat_imgs,output_image
+
+def main():
+    args = build_argparser().parse_args()
 
+    if args.auto:
+        # Command-line inpaining for just one image
+        concat_image,result = inpaintRandomHoles(args)
+        if args.output != "":
+            cv2.imwrite(args.output,result)
+        if not args.no_show:
+            cv2.imshow('Image Inpainting Demo', concat_image)
+            cv2.waitKey(0)
+    else:
+        # Inpainting with GUI
+        InpaintingGUI(loadImage(args.input),args.model, args.device).run()
```

Comment:
You should load the img in `InpaintingGUI`'s constructor like for `inpaintRandomHoles()`.

Reply:
Image is loaded in main function now. Once for both parts - no matter which one is selected. Let's keep it this way